src/main/kotlin/org/jetbrains/research/testspark/actions/llm/LLMSampleSelectorFactory.kt

```diff
@@ -172,4 +180,55 @@ class LLMSampleSelectorFactory(private val project: Project) : PanelFactory {
             testSamplePanelFactory.enabledComponents(isEnabled)
         }
     }
+
+    /**
+     * Retrieves a list of test samples from the given project.
+     *
+     * @return A list of strings, representing the names of the test samples.
+     */
+    private fun collectTestSamples(project: Project, testNames: MutableList<String>, initialTestCodes: MutableList<String>) {
+        val projectFileIndex: ProjectFileIndex = ProjectRootManager.getInstance(project).fileIndex
+        val javaFileType: FileType = FileTypeManager.getInstance().getFileTypeByExtension("java")
```

Comment:
Do we create an implementation for Java only right now? In the future, we will need to refactor it to integrate Kotlin support.

Reply:
This implementation focuses exclusively on the Java language. All future refactorings and extensions to other languages will be in separate PRs. Kate will implement some manager for interaction with other languages